build-scripts/build-custom-app.sh

```diff
@@ -0,0 +1,234 @@
+};
+
+export default Methodology;
+
+EOM
+
+# Define the content for homepage.tsx
+read -r -d '' HOMEPAGE_CONTENT <<EOM
+
+import React from "react";
+// generic layer component
+import Layout from "@components/layouts/LandingPage";
+
+import { Hero } from "@${custom_app_name}/components/Hero";
+
+type TProps = {
+  handleSearchInput: (term: string, filter?: string, filterValue?: string) => void;
+  searchInput: string;
+};
+
+const LandingPage = ({ handleSearchInput, searchInput }: TProps) => {
+  return (
+    <Layout title="">
+      <main id="main" className="flex flex-col flex-1">
+        <div className="bg-cclw-dark">
+          <Hero handleSearchInput={handleSearchInput} searchInput={searchInput} />
+        </div>
+      </main>
+    </Layout>
+  );
+};
+
+export default LandingPage;
+
+
+EOM
+
+# Define the content for the hero.
+
+read -r -d '' HERO_CONTENT <<EOF
+
+import LandingSearchForm from "./LandingSearchForm";
+
+
+type TProps = {
+  handleSearchInput: (term: string, filter?: string, filterValue?: string) => void;
+  searchInput: string;
+};
+
+export const Hero = ({ handleSearchInput, searchInput }: TProps) => {
+  return (
+    <div className="pb-6 text-white pt-[28px] sm:pt-[48px] md:pt-[80px] lg:pt-[100px] xl:pt-[140px]">
+      <div className="container">
+        <div className="flex flex-col items-center justify-center mb-6">
+		  <p className="font-medium tracking-slight text-lg lg:text-3xl" data-cy="intro-message">
+          ${custom_app_name} custom app
+        </p>
+        </div>
+        <div className="max-w-screen-md mx-auto mt-6">
+          <LandingSearchForm handleSearchInput={handleSearchInput} placeholder="Search full text of any document" input={searchInput} />
+        </div>
+      </div>
+    </div>
+  );
+};
+
+
+EOF
+
+# Define the content for the routes.json
+
+read -r -d '' ROUTES_JSON_CONTENT <<EOM
+
+[
+  {
+    "title": "Methodology",
+    "path": "methodology",
+    "contentPath": "methodology"
+  }
+]
+
+EOM
+
+# Define the content for the index.ts file
+
+read -r -d '' INDEX_TS_CONTENT <<EOM
+
+export { default as Analytics } from "./Analytics";
+export { default as Hero } from "./Hero";
+
+EOM
+
+# Define the path to the themes directory
+themes_path="themes"
+
+# Define the path to the e2e directory
+e2e_path="e2e"
+
+# Check if the themes directory exists
+if [[ ! -d ${themes_path} ]]; then
+	echo "The themes directory does not exist. Creating it now..."
+	mkdir "${themes_path}"
+fi
+
+# Define the full path to the new directory
+new_dir="${themes_path}/${custom_app_name}"
+# Define the path to the subdirectories
+components_dir="${new_dir}/components"
+layouts_dir="${new_dir}/layouts"
+pages_dir="${new_dir}/pages"
+styles_dir="${new_dir}/styles"
+tests_dir="${new_dir}/tests"
+e2e_test_dir="${e2e_path}/cypress/e2e/${custom_app_name}"
+
+# Check if the directory already exists
+if [[ -d ${new_dir} ]]; then
+	echo "You are trying to duplicate a Custom App, ${new_dir} already exists."
```

Comment:
Maybe we could let the user know what they need to do if they want to overwrite the existing custom app at this location?
```suggestion
	echo "You are trying to duplicate a Custom App, ${new_dir} already exists. Please delete this folder and its contents and try again if you want to create a new custom app at this path"
```